sentry_streams/sentry_streams/examples/transformer.py

```diff
@@ -53,17 +47,16 @@ def merge(self, other: Self) -> Self:
 
 reduce_window = SlidingWindow(window_size=timedelta(seconds=6), window_slide=timedelta(seconds=2))
 
-pipeline = (
-    streaming_source(
-        name="myinput",
-        stream_name="events",
-    )
-    .apply("mymap", Map(function=parse))
-    .apply("myfilter", Filter(function=filter_not_event))
-    .apply("myreduce", Reducer(reduce_window, TransformerBatch))
-    .apply("serializer", Map(function=serialize_msg))
-    .sink(
-        "kafkasink2",
-        stream_name="transformed-events",
-    )  # flush the batches to the Sink
-)
+pipeline = streaming_source(name="myinput", stream_name="ingest-metrics")
+
+chain2 = pipeline.apply("parser", Parser(msg_type=IngestMetric, deserializer=json_parser))
```

Comment:
I split up the pipeline like this because you can see the inferred type hints in an editor